.github/actions/inject-issue-id-action/action.yaml

```diff
@@ -0,0 +1,98 @@
+    # String will be added, output variable set true
+    description: "Whether to inject an issue-id if NOT found"
+    type: boolean
+    required: false
+    default: true
+
+outputs:
+  PRESENT:
+    description: "Whether the required ticket string is present"
+    value: ${{ steps.status.outputs.present }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ github.head_ref }}
+
+    - name: "Get GIT commit message body"
+      id: commit-message
+      # yamllint disable rule:line-length
+      uses: lfit/releng-reusable-workflows/.github/actions/git-commit-message-action@main
+
+    - name: "Check commit message for ticket/string"
+      shell: bash
+      run: |
+        # Check commit message for ticket/string
+        echo "# Checking Commit Message for JIRA Ticket" >> "$GITHUB_STEP_SUMMARY"
+
+        COMMIT_MESSAGE="${{ env.COMMIT_MESSAGE }}"
+        if [ -z "$COMMIT_MESSAGE" ]; then
+          echo "Commit message body was empty/unpopulated ❌"; exit 1
+        fi
+        if [[ "$COMMIT_MESSAGE" =~ "Issue-ID: " ]]; then
+          echo "Ticket string found in the pull request ✅" >> "$GITHUB_STEP_SUMMARY"
+          echo "Ticket string found in the pull request ✅"
+          echo "PRESENT=true" >> "$GITHUB_ENV"
+        else
+          echo "Ticket string NOT found in the pull request ❌" >> "$GITHUB_STEP_SUMMARY"
+          echo "Ticket string NOT found in the pull request ❌"
+          echo "PRESENT=false" >> "$GITHUB_ENV"
+        fi
+
+    - name: "Set key to use for JSON lookup"
+      if: env.PRESENT == 'false' && inputs.INJECT == 'true'
+      shell: bash
+      run: |
+        # Set key to use for JSON lookup
+        ACTOR="${{ github.actor }}"
+        ACTOR_ID="${{ github.actor_id }}"
+        echo "Using GitHub actor as lookup key: $ACTOR [$ACTOR_ID] 🔑"
+        echo "key=$ACTOR" >> "$GITHUB_ENV"
+
+    - name: "Get ticket from JSON lookup table"
+      if: env.PRESENT == 'false' && inputs.INJECT == 'true'
+      uses: lfit/releng-reusable-workflows/.github/actions/json-key-value-lookup-action@main
+      with:
+        json: ${{ inputs.issue_id_lookup_json }}
+        key: ${{ env.key }}
+
+    - name: "Configure GIT user for commit"
+      if: env.PRESENT == 'false' && inputs.INJECT == 'true'
+      uses: lfit/releng-reusable-workflows/.github/actions/git-configure-action@main
+
+    - name: "Adding JIRA ticket reference to commit message"
+      id: inject
+      # yamllint disable-line rule:line-length
+      if: env.PRESENT == 'false' && inputs.INJECT == 'true' && (github.event.pull_request.head.ref != github.ref_name) && (github.event.pull_request.head.ref != github.event.pull_request.base.ref)
+      shell: bash
+      run: |
+        # Adding JIRA ticket reference to commit message
+        git commit --amend -m "${{ env.COMMIT_MESSAGE }}" -m "Issue-ID: ${{ env.value }}" --no-verify --no-edit
```

Comment:
Instead of -m use the `--trailer` git-commit option, since that add directly to the git commit footer without messing around with the commit message format.

Reply:
As mentioned above, the commit footer should not be hard coded to `Issue-ID` as some of our projects use a different footer syntax. This should be a lookup value.

Reply:
Wasn't aware of this feature, and it's nice that it's available, that will be a much better solution.

Reply:
Implemented and now tested here:
https://github.com/ModeSevenIndustrialSolutions/test-issue-id/pull/51/commits/a11cc0c0abcd1ad1df74f7cfca9b4bb481e1fef1